rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -160,95 +194,105 @@ struct MetaDataBatch
+    std::vector<uint>& get_image_id_batch() { return _info_batch._img_ids; }
+    std::vector<std::string>& get_image_names_batch() {return _info_batch._img_names; }
+    ImgSizes& get_img_sizes_batch() { return _info_batch._img_sizes; }
+    MetaDataDimensionsBatch& get_metadata_dimensions_batch() { return _metadata_dimensions; }
+    int get_batch_object_count() { return _total_objects_count; }
+    void reset_objects_count() {
         _total_objects_count = 0;
     }
     void increment_object_count(int count) { _total_objects_count += count; }
-    int get_batch_object_count() { return _total_objects_count; }
-    MetaDataDimensionsBatch& get_metadata_dimensions_batch() { return _metadata_dimensions; }
 protected:
-    std::vector<int> _label_id = {}; // For label use only
-    std::vector<BoundingBoxCords> _bb_cords = {};
-    std::vector<BoundingBoxCords_xcycwh> _bb_cords_xcycwh = {};
-    std::vector<BoundingBoxLabels> _bb_label_ids = {};
-    ImgSizes _img_sizes = {};
-    std::vector<size_t> _buffer_size;
-    int _total_objects_count = 0;
     MetaDataDimensionsBatch _metadata_dimensions;
+    MetaDataInfoBatch _info_batch;
+    int _total_objects_count = 0;
 };
 
 struct LabelBatch : public MetaDataBatch
 {
     void clear() override
     {
-        _label_id.clear();
+        for (int i = 0; i < _label_ids.size(); i++) {
+            _label_ids[i].clear();
+        }
+        _label_ids.clear();
         _buffer_size.clear();
         _total_objects_count = 0;
     }
     MetaDataBatch&  operator += (MetaDataBatch& other) override
     {
-        _label_id.insert(_label_id.end(), other.get_label_batch().begin(), other.get_label_batch().end());
+        _label_ids.insert(_label_ids.end(), other.get_label_batch().begin(), other.get_label_batch().end());
         return *this;
     }
     void resize(int batch_size) override
     {
-        _label_id.resize(batch_size);
+        _label_ids.resize(batch_size);
     }
     int size() override
     {
-        return _label_id.size();
+        return _label_ids.size();
     }
     std::shared_ptr<MetaDataBatch> clone() override
     {
         return std::make_shared<LabelBatch>(*this);
     }
-    explicit LabelBatch(std::vector<int>& labels)
+    explicit LabelBatch(std::vector<Labels>& labels)
     {
-        _label_id = std::move(labels);
+        _label_ids = std::move(labels);
     }
     LabelBatch() = default;
     void copy_data(std::vector<void*> buffer) override
     {
         if(buffer.size() < 1)
             THROW("The buffers are insufficient") // TODO -change
-        memcpy((int *)buffer[0], _label_id.data(), _label_id.size() * sizeof(int));
+        auto labels_buffer = (int *)buffer[0];
+        for (int i = 0; i < _label_ids.size(); i++) {
+            memcpy(labels_buffer, _label_ids[i].data(), _label_ids[i].size() * sizeof(int));
+            labels_buffer += _label_ids[i].size();
+        }
     }
     std::vector<size_t>& get_buffer_size() override
```

Comment:
@fiona-gladwin Is this required?

Reply:
Yes @fiona-gladwin. This was called by ring buffer before copy data called